hbase-client/src/main/java/org/apache/hadoop/hbase/client/TableDescriptorBuilder.java

```diff
@@ -1335,11 +1335,14 @@ public boolean hasRegionMemStoreReplication() {
      * @return the modifyable TD
      */
     public ModifyableTableDescriptor setRegionMemStoreReplication(boolean memstoreReplication) {
-      setValue(REGION_MEMSTORE_REPLICATION_KEY, Boolean.toString(memstoreReplication));
-      // If the memstore replication is setup, we do not have to wait for observing a flush event
+      ModifyableTableDescriptor returnDesc =
+          setValue(REGION_MEMSTORE_REPLICATION_KEY, Boolean.toString(memstoreReplication));
+      // If the memstore replication not setup, we do not have to wait for observing a flush event
       // from primary before starting to serve reads, because gaps from replication is not applicable
-      return setValue(REGION_REPLICA_WAIT_FOR_PRIMARY_FLUSH_CONF_KEY,
-              Boolean.toString(memstoreReplication));
+      if (!memstoreReplication) {
```

Comment:
This is a behavior change? If memstoreReplication is true, we will not set REGION_REPLICA_WAIT_FOR_PRIMARY_FLUSH_CONF_KEY to true?

Reply:
@Apache9 , yes, the fix of the problem described on jira is here, if we explicitly set memstoreReplication to true at table level, we do not change the  REGION_REPLICA_WAIT_FOR_PRIMARY_FLUSH_CONF_KEY, keep its original config(the default value is true). If REGION_REPLICA_WAIT_FOR_PRIMARY_FLUSH_CONF_KEY is false and memstoreReplication is true, the secondary region is enabled for read when open.

Reply:
I suggest we do the logic outside TableDescriptorBuilder. It is confusing...

Reply:
@Apache9 ,  you mean it is better to remove the  additional 
` setValue(REGION_REPLICA_WAIT_FOR_PRIMARY_FLUSH_CONF_KEY,Boolean.toString(memstoreReplication))` from  `TableDescriptorBuilder.setRegionMemStoreReplication` , yes , these two configuration items are not necessarily related，I look the original patch for HBASE-13063, added this additional 
` setValue(REGION_REPLICA_WAIT_FOR_PRIMARY_FLUSH_CONF_KEY,Boolean.toString(memstoreReplication))` is just for test.

Reply:
@Apache9 , I have remove the logic into `HRegionServer.triggerFlushInPrimaryRegion`